### PR TITLE
Add a hint to the revoke call.

### LINKED ISSUE #229: `revoke()` should take a `hint` that is independent from the `accountId`
The revocation endpoint takes "hints" that are different than the "account_id" that it passed to FedCM during the "accounts_endpoint". So, I think we should expose some form of parameter in the revoke method to allow that to be passed to the HTTP endpoint.

Here is a draft of what I was imagining, but open to any other formulation that allows this to be done:

```js
const cred = await navigator.credentials.get({
  federated: {
    providers: [{
      url: "idp.example",
      clientId: "123",
      // no hint needed here, see below
    }]
  }
});
cred.revoke({
  hint: "user@google.com"
})
```

### PR DESCRIPTION
**Pull request by cbiesinger**
Fixes #229

***
<a href="https://pr-preview.s3.amazonaws.com/cbiesinger/WebID/pull/232.html" title="Last updated on Mar 14, 2022, 7:25 PM UTC (22a4064)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/fedidcg/FedCM/232/b3ae0c8...cbiesinger:22a4064.html" title="Last updated on Mar 14, 2022, 7:25 PM UTC (22a4064)">Diff</a>

### CODE REVIEW

File: spec/index.bs [hunk 1]
```diff
@@ -867,8 +867,12 @@ It will also contain the following parameters in the request body `application/x
     :: The account identifier for the user (e.g. email).
     :   <dfn>client_id</dfn>
     :: The [=RP=]'s client id
+    :   <dfn>hint</dfn>
+    :: The hint provided to the JS call.
```
Comment (dj2) [inline spec/index.bs:871]: Should we remove `account_id` and just have `hint`? When would I use one over the other?

File: spec/index.bs [hunk 2]
```diff
@@ -1339,7 +1350,7 @@ path: img/mock31.svg
 
 <div class=example>
 ```js
-await FederatedCredential.logoutRPs([{
+await FederatedCredential.logoutRps([{
```
Comment (dj2) [inline spec/index.bs:1353]: This seems unrelated to this PR.

File: spec/index.bs (hunk 1 above)
Comment (cbiesinger) [reply spec/index.bs:869]: Yeah, probably better to remove it until we know we need it (I was thinking some IDPs may just use their account IDs instead of a different hint, but they can just pass the id as the argument)

File: spec/index.bs (hunk 2 above)
Comment (cbiesinger) [reply spec/index.bs]: Done

File: spec/index.bs (hunk 1 above)
Comment (cbiesinger) [reply spec/index.bs:869]: To be clear, I changed it to just hint and made that argument required.

**Review by dj2 — APPROVED** [review]

File: spec/index.bs [hunk 3]
```diff
@@ -1275,10 +1275,13 @@ method:
 <xmp class=idl>
 [Exposed=Window, SecureContext]
 partial interface FederatedCredential {
-  Promise<undefined> revoke();
+  Promise<undefined> revoke(USVString hint);
```
Comment (samuelgoto) [inline spec/index.bs:1278]: Should this take a dictionary instead of a string? Don't we anticipate revoke to need more parameters as we go along?

File: spec/index.bs [hunk 4]
```diff
@@ -291,7 +291,7 @@ could be implemented.
   async function revoke() {
     const credential = getFederatedCredential();
     // This never prompts, rejects the promise in case !credential[“consented”]
-    return credential.revoke();
+    return credential.revoke(credential.id);
```
Comment (samuelgoto) [inline spec/index.bs:294]: isn't `credential.id` precisely the account identifier which isn't useful? that is, isn't the case that, in practice, this isn't going to be thing to be pass? wouldn't a better example be:

```js
return credential.revoke(email);
```

Comment (samuelgoto) [inline spec/index.bs:294]: isn't `credential.id` precisely the account identifier which isn't useful? that is, isn't the case that, in practice, this isn't going to be thing to be pass? wouldn't a better example be:

```js
return credential.revoke(email);
```

Comment (cbiesinger) [reply spec/index.bs:294]: Well, I figured it would work for some IDPs, and in another example I did use an email address

File: spec/index.bs (hunk 3 above)
Comment (cbiesinger) [reply spec/index.bs:1278]: To summarize the discussion elsewhere: If we later on find a need for another parameter, we can easily change this function to take string|dictionary
